tests/cache/tests.py

```diff
@@ -585,15 +585,23 @@ def func(key, *args):
             with warnings.catch_warnings(record=True) as w:
                 warnings.simplefilter("always")
                 # memcached does not allow whitespace or control characters in keys
-                cache.set('key with spaces', 'value')
+                cache.set('key with spaces and 清', 'value')
                 self.assertEqual(len(w), 1)
                 self.assertIsInstance(w[0].message, CacheKeyWarning)
+                # On Python 2 ``warnings.warn()`` expects a ``message`` that
+                # can be coerced to a ``str``. If this fails -
+                # ``warnings.warn()`` will also fail.
+                str(w[0].message.args[0])
```

Comment:
Is there a possibility to use `assertEqual` to verify the message instead of this?